storage_delegation/main.tf

```diff
@@ -1,3 +1,9 @@
+module "crn_parser" {
+  source  = "terraform-ibm-modules/common-utilities/ibm//modules/crn-parser"
+  version = "1.1.0"
+  crn     = var.cos_kms_key_crn
```

Comment:
`cos_kms_key_crn` is an optional input, so your going to need a count around this module call so its only called if a value is passed

Reply:
kms_key_crn is now passed to parser module by reading it from data source which will be not null.